mmrotate/datasets/hrsc.py

```diff
@@ -1,266 +1,178 @@
 # Copyright (c) OpenMMLab. All rights reserved.
 import os.path as osp
 import xml.etree.ElementTree as ET
-from collections import OrderedDict
+from typing import List, Union
 
 import mmcv
 import numpy as np
-from mmcv import print_log
-from mmengine.dataset import BaseDataset
-from PIL import Image
+from mmdet.datasets import XMLDataset
 
-from mmrotate.core import eval_rbbox_map, obb2poly_np, poly2obb_np
-from .builder import ROTATED_DATASETS
+from mmrotate.core import obb2poly_np
+from mmrotate.registry import DATASETS
 
 
-@ROTATED_DATASETS.register_module()
-class HRSCDataset(BaseDataset):
+@DATASETS.register_module()
+class HRSCDataset(XMLDataset):
```

Comment:
This class seems to overwrite all main methods of XMLDataset. Is it necessary to inherit from XMLDataset?

Reply:
Good suggestion!